javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -223,6 +219,21 @@ final class Matcher private[regex] (
 
   def hasAnchoringBounds(): Boolean = true
   //def useAnchoringBounds(b: Boolean): Matcher
+
+  private[regex] def string(): String = inputstr
+  private[regex] def getRegex(): js.RegExp = regexp
+
+  // For performance reasons, if we don't use the positions
+  private var cache: Map[Int, Int => Int] = Map()
+
+  /** Returns a mapping from the group number to the respective start position */
+  private[regex] def startOf: (Int => Int) = {
+    cache.getOrElse(start(), {
+      val groupPositions = computeCurrentGroupPositions()
+      cache += start() -> groupPositions
+      groupPositions
+    })
```

Comment:
I simplified the process, and removed the dependency to start() so that it invalidates the cache `Option[Int => Int]` every time there is a call to `find`, `reset` or `usePattern`
I also renamed StartHack to GroupStartMap.